circle.yml

```diff
@@ -0,0 +1,61 @@
+---
+version: 2
+
+buildSteps: &buildSteps
+  - checkout
+  - restore_cache:
+      keys:
+        - dependency-cache-{{ checksum "package.json" }}
+        # fallback to the latest cache if no match is found
+        - v1-dependencies-
+  - run:
+      name: install-npm
+      command: npm install
+  - save_cache:
+      key: v1-dependencies-{{ checksum "package.json" }}
+      paths:
+        - node_modules
+  - run:
+      name: test
+      command: npm test
+
+jobs:
+  "node-6":
+    docker:
+      - image: circleci/node:6
+    steps: *buildSteps
+  "node-8":
+    docker:
+      - image: circleci/node:8
+    steps: *buildSteps
+
+  deploy:
+    # For this to work NPM_TOKEN must be set for the account used for publishing
+    docker:
+      - image: circleci/node:6
+    steps:
+      - run:
+          name: Login to npm
+          command: |
+            echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+            npm whoami
+      - deploy:
+          name: Deploy to npm
+          command: npm publish
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - "node-6"
+      - "node-8"
+      - deploy:
+          # Deploy passing builds if they're tagged with a version
+          requires:
+            - "node-6"
+            - "node-8"
+          filters:
+            tags:
+              only: /^v\d+\.\d+\.\d+$/
+            branches:
+              ignore: /.*/
```

Comment:
Is it worth also restricting this to master, for safety?

Reply:
You can't, sadly. Git tags aren't explicitly related to branches, they're independent standalone references. This line is just ignoring every single branch, so it doesn't get deployed for master _and_ the tagged commit.

What we could try to say to get that sort of behaviour is "does this tag refer to a commit that's also on master", but that's the best you can theoretically get in terms of relating the two, and circle (or travis or anything else) doesn't seem to have any kind of built-in support for filtering like that, so it's a little tricky to do. I'd like to extend to do that kind of thing eventually, but I don't _think_ it's strictly necessary for a first implementation. I can look into it more if you disagree though.